internal/firewall/resource.go

```diff
@@ -117,7 +117,15 @@ func Resource() *schema.Resource {
 								Type:             schema.TypeString,
 								ValidateDiagFunc: validateIPDiag,
 								StateFunc: func(i interface{}) string {
-									return strings.ToLower(i.(string))
+									ipS := strings.ToLower(i.(string))
+									if !strings.Contains(ipS, "/") {
+										if strings.Contains(ipS, ":") {
+											ipS += "/64"
+										} else if strings.Contains(ipS, ".") {
+											ipS += "/32"
+										}
+									}
+									return ipS
```

Comment:
Using string manipulation for this feels wrong and brittle to me.

Can we instead use the [`net` package](https://pkg.go.dev/net) and its methods and structs `ParseCIDR`, `ParseIP`, `IPNet` to implement this?

Some quick pseudo code:

```go
_, ipnet, err := net.ParseCIDR(i.(string))
if err != nil {
  ip := net.ParseIP(i.(string))
  if ip == nil {
    // No CIDR or IP
    return
  }
  if ip.To4() != nil {
    // IPv4
    ipnet = &net.IPNet{
      IP: ip,
      IPMask: net.CIDRMask(32, 32)
    }
  } else {
     // If To4 returns nil, IP is not IPv4 => IPv6
    ipnet = &net.IPNet{
      IP: ip,
      IPMask: net.CIDRMask(64, 128)
    }
  }

  return ipnet.String()
}
```